pkg/dio/writer.go

```diff
@@ -141,39 +120,35 @@ func (t *multiWriter) Close() error {
 	return nil
 }
 
-// TODO: add a ReadFrom for more efficient copy
-
-// Remove doesn't return an error if element isn't found as the end result is
-// identical
-func (t *multiWriter) Remove(writer io.Writer) {
-	t.mutex.Lock()
-	defer t.mutex.Unlock()
+func (t *multiWriter) remove(writer io.Writer) {
+	wTmp := make([]io.Writer, 0, len(t.writers))
 
 	if verbose {
 		log.Debugf("[%p] removing writer %p - currently %d writers", t, writer, len(t.writers))
 	}
-	for i, w := range t.writers {
-		if w == writer {
-			t.writers = append(t.writers[:i], t.writers[i+1:]...)
-			if verbose {
-				log.Debugf("[%p] removed writer - now %d writers", t, len(t.writers))
-
-				for i, w := range t.writers {
-					log.Debugf("[%p] Writer %d [%p]", t, i, w)
-				}
-			}
-			break
+	for _, w := range t.writers {
+		if w != writer {
+			wTmp = append(wTmp, w)
 		}
 	}
+	if len(t.writers) != len(wTmp) {
+		log.Debugf("[%p] removed writer - now %d writers", t, len(wTmp))
+	}
+	t.writers = wTmp
+}
+
+// Remove doesn't return an error if element isn't found as the end result is
+// identical
+func (t *multiWriter) Remove(writer io.Writer) {
+	t.mutex.Lock()
+	t.remove(writer)
+	t.mutex.Unlock()
 }
 
 // MultiWriter extends io.MultiWriter to allow add/remove of writers dynamically
 // without disrupting existing writing
 func MultiWriter(writers ...io.Writer) DynamicMultiWriter {
-	w := make([]io.Writer, len(writers))
-	copy(w, writers)
-	t := &multiWriter{writers: w}
-
+	t := &multiWriter{writers: writers}
```

Comment:
huh, interesting. Why is this copy unnecessary?

Reply:
I guess the answer to that could just be "because it is, and it wasn't needed in the first place"

Reply:
yup